### PR TITLE
Anthos newsletter 1: Why we chose Anthos

### PR DESCRIPTION
**Pull request by esphen**
This is the first newsletter in a series of 3 on our new tech blog.

The idea is that we can build on this later to talk about how we use Anthos and our learnings with Anthos in later posts.

Content review much appreciated! 🙂

### CODE REVIEW

**Review by SnorreSelmer — CHANGES_REQUESTED** [review]
Found these typos, other than that it looks very good!

Line 33: "tialing" > "trialing"
Line 64: "assumtions" > "assumptions"

**Review by SnorreSelmer — APPROVED** [review]
:+1:

File: blog/2023-09-11-anthos-1.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+This has been something we've worked on for a long time, and still are. After a
+long time of working with the stakeholders in the organization, we eventually
+built a cloud strategy, which in simple terms stated that we would prefer 
+SaaS-products over hosting things ourselves, and that we would gradually move
+our workloads to the cloud. 
+
+This cloud strategy, however, came too late for us on SKIP. We had already
+done most of the work on our on-premise platform, building on the assumptions the
+organization held at the time, which was "cloud is scary" and "we need to run
```
Comment (hustuss) [inline blog/2023-09-11-anthos-1.md:65]: Maybe rephrase this to take some sting out of it? Something like "building on the assumptions the organization held at the time, which was that we met our needs through existing infrastructure and had no need for cloud technology"

**Review by hustuss — COMMENTED** [review]
Just left a minor comment, not sure if I'm too paranoid or not...

File: blog/2023-09-11-anthos-1.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+Were you to take the time machine back a few years, you would see Kartverket as a
+traditional enterprise with a lot of knowledge and experience in running
+on-premise workloads. This knowledge served us well, but also slightly held us
+back in terms of our imagination. We knew that there had to be a better way,
+but our enterprise was simply not mature enough to adopt a pure cloud strategy.
+The fear of the unknown cloud weighed heavily on many people, and therefore few
+people wanted to take the risk of moving to the cloud.
+
+This has been something we've worked on for a long time, and still are. After a
```
Comment (lislei) [inline blog/2023-09-11-anthos-1.md:57]:
```suggestion
This is something we've worked on for a long time, and still are. After a
```

File: blog/2023-09-11-anthos-1.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+Anthos is Google's solution to multicloud. It's a product portfolio where the
+main product is GKE (Google Kubernetes Engine) on-premise. Using GKE on-prem
+you can run Kubernetes clusters on-premise and manage them from the same
+control plane in Google Cloud, as if they were proper cloud clusters.
+
+In fact, Anthos is truly multi-cloud. That means you can deploy Anthos
+clusters to GKE and on-prem, but also AWS and Azure. On other cloud platforms
+it uses the provider's Kubernetes distribution like AKS, but you can still
+manage it from GKE anongside your other clusters.
```
Comment (lislei) [inline blog/2023-09-11-anthos-1.md:126]:
```suggestion
manage it from GKE alongside your other clusters.
```

File: blog/2023-09-11-anthos-1.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+firewalls. 
+
+All traffic that flows between microservices on the mesh is also automatically 
+encrypted with mTLS.
+
+### Anthos Config Managment (ACM)
+
+A way to sync git repos into a running cluster. Think GitOps here. Build a repo
+containing all your Kubernetes manifests and sync them into your cluster, making
+cluster maintinance easier.
```
Comment (lislei) [inline blog/2023-09-11-anthos-1.md:145]:
```suggestion
cluster maintenance easier.
```

File: blog/2023-09-11-anthos-1.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+`kubectl get pods` or `kubectl logs` (including `-f`). It does not support
+`port-forward`, `exec` or `run`.
+
+## Summary
+
+As you can see, the above tools gives us a lot of benefits.
+
+- Combined with the power of Google Cloud and
+Terraform, they give us a good combination of flexibility through cloud services
+- Ease the maintainance by using the tools that Anthos and Terraform supplies us
```
Comment (lislei) [inline blog/2023-09-11-anthos-1.md:173]:
```suggestion
- Ease the maintenance by using the tools that Anthos and Terraform supply us
```
supplies is not a verb. Commonly used is to leverage something - perhaps that’s even better here?

File: blog/2023-09-11-anthos-1.md [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+
+In Kartverket we have an ambition to adopt cloud native technologies. There's
+thousands of ways to do this, and after trialing a couple of alternative
+solutions, including running plain Kubernetes and VMWare Tanzu, we decided to go
+with Anthos. Anthos is a platform that allows us to run Kubernetes clusters
+on-premise and in the cloud, and manage them from a single pane of glass. We
+have been running Anthos in production for a while now, at least long enough to
+be able to share our thoughts.
+
+This newsletter is part of a three-part series newsletter about Anthos in
```
Comment (lislei) [inline blog/2023-09-11-anthos-1.md:40]:
```suggestion
This newsletter is the first of three part series about Anthos in
```
Avoid repeating of the word newsletter

File: blog/2023-09-11-anthos-1.md [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+![Anthos logo](img/anthos-2.png)
+
+Anthos is Google's solution to multicloud. It's a product portfolio where the
+main product is GKE (Google Kubernetes Engine) on-premise. Using GKE on-prem
+you can run Kubernetes clusters on-premise and manage them from the same
+control plane in Google Cloud, as if they were proper cloud clusters.
+
+In fact, Anthos is truly multi-cloud. That means you can deploy Anthos
+clusters to GKE and on-prem, but also AWS and Azure. On other cloud platforms
+it uses the provider's Kubernetes distribution like AKS, but you can still
```
Comment (lislei) [inline blog/2023-09-11-anthos-1.md:125]: I had to look up AKS - perhaps explain that acronym here as well :)

**Review by lislei — COMMENTED** [review]
Interesting read Espen - I enjoyed it!

File: blog/2023-09-11-anthos-1.md [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+or how to have some workloads running on-premise and some in the the cloud in a
+good way. 
+
+In this newsletter-series we will share some of our experiences solving this
+issue by running GKE on-prem in Kartverket, including the reasons we went with
+Anthos and pros and cons we have experienced so far.
+
+<!--truncate-->
+
+In Kartverket we have an ambition to adopt cloud native technologies. There's
```
Comment (evenh) [inline blog/2023-09-11-anthos-1.md:32]:
```suggestion
At [Kartverket](https://www.kartverket.no/en) we have an ambition to adopt cloud native technologies. There's
```

File: blog/2023-09-11-anthos-1.md [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+The fear of the unknown cloud weighed heavily on many people, and therefore few
+people wanted to take the risk of moving to the cloud.
+
+This has been something we've worked on for a long time, and still are. After a
+long time of working with the stakeholders in the organization, we eventually
+built a cloud strategy, which in simple terms stated that we would prefer 
+SaaS-products over hosting things ourselves, and that we would gradually move
+our workloads to the cloud. 
+
+This cloud strategy, however, came too late for us on SKIP. We had already
```
Comment (evenh) [inline blog/2023-09-11-anthos-1.md:63]:
```suggestion
This cloud strategy, however, came too late for us on <abbr title="Statens Kartverk Infrastructure Platform">SKIP</abbr>. We had already
```

File: blog/2023-09-11-anthos-1.md
[the remark this answers is not in the document]
Comment (evenh) [reply blog/2023-09-11-anthos-1.md:1]: Also consider that the audience may come from sites like news.ycombinator.com, Reddit or other international sites.

File: blog/2023-09-11-anthos-1.md [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+Because of these limitations, a pure cloud strategy is not considered to be a
+good fit for us.
+
+A hybrid cloud, however, can give us the scalability and flexibility of the
+cloud, while still allowing us to run some of our systems on-prem, with the
+experience being more or less seamless for the developers.
+
+## Why we chose Anthos
+
+After some disastrous issues with our previous hybrid cloud PoC, VMWare Tanzu,
```
Comment (evenh) [inline blog/2023-09-11-anthos-1.md:87]: As we're calling out a vendor here, maybe share some details on why the experience was bad?

File: blog/2023-09-11-anthos-1.md (hunk 7 above)
Comment (evenh) [reply blog/2023-09-11-anthos-1.md:125]: I think AKS is well known in platform engineering circles.

File: blog/2023-09-11-anthos-1.md [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+
+- We had a decent pool of knowledge in GCP compared to AWS and Azure at the time
+- Some very well established platform teams in the public sector were also using
+  GCP, which meant it would be easier to share work and learnings
+- Anthos and GCP seemed to offer a good developer experience, which for us as a
+  platform team is of paramount importance
+- A provider like Google is well established in the cloud space (especially
+  Kubernetes), and would have a fully featured, stable and user friendly product
+
+SKIP ran the Anthos PoC over a few months. This took us all the way from
```
Comment (evenh) [inline blog/2023-09-11-anthos-1.md:100]: Consider saying something about that this was/is on-prem only and was done in concert with internal network and infrastructure engineers

File: blog/2023-09-11-anthos-1.md [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+provisioning clusters and networking, to iterating on tools and docs and finally
+onboarding an internal product team on the platform. Once we felt we had learned
+what we could from the PoC, we gathered thoughts from the product team,
+infrastructure team and of course the SKIP platform team.
+
+The results were unanimous. All the participants lauded the GCP user interfaces that
+allowed visibility into running workloads, as well as the new self-service
+features that came with it. Infrastructure complimented the installation scripts
```
Comment (evenh) [inline blog/2023-09-11-anthos-1.md:108]:
```suggestion
features that came with it. Infrastructure engineers complimented the installation scripts
```

File: blog/2023-09-11-anthos-1.md [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+A networking solution based on [Istio](http://istio.io). This is sort of the
+backbone of the hybrid features of Anthos, as provided you've configured a
+hybrid mesh it allows applications deployed to the cloud to communicate with
+on-premise workloads automatically and without manual steps like opening
+firewalls. 
+
+All traffic that flows between microservices on the mesh is also automatically 
+encrypted with mTLS.
+
+### Anthos Config Managment (ACM)
```
Comment (evenh) [inline blog/2023-09-11-anthos-1.md:141]: Is there a dependency on Kustomize, or just the way it was used internally?

File: blog/2023-09-11-anthos-1.md [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+firewall. From a user experience standpoint this is quite useful, as devs
+will be logged in to GCP using two factor authentication, and the same strong
+authentication allows you to access kubernetes on-premise.
+
+Connect Gateway also integrates with GCP groups, enabling RBAC in Kubernetes
+to be assigned to groups instead of manually administered lists of users.
+
+Currently the connect gateway only supports stateless requests, for example
+`kubectl get pods` or `kubectl logs` (including `-f`). It does not support
+`port-forward`, `exec` or `run`.
```
Comment (evenh) [inline blog/2023-09-11-anthos-1.md:165]: I think we should underline that this has been a painpoint for us, for debuggability.

Comment (esphen) [reply blog/2023-09-11-anthos-1.md:165]: I was planning on digging into those issues in the next newsletters :slightly_smiling_face:

File: blog/2023-09-11-anthos-1.md (hunk 10 above)
Comment (esphen) [reply blog/2023-09-11-anthos-1.md:87]: That's a whole blog of it's own 😄 I'll redact the name for now
